partner_relations/model/res_partner_relation.py

```diff
@@ -0,0 +1,366 @@
+#
+##############################################################################
+
+from openerp import osv, models, fields, api, exceptions, _
+
+from . import get_partner_type
+
+
+class ResPartnerRelation(models.Model):
+    '''Model res.partner.relation is used to describe all links or relations
+    between partners in the database.
+
+    In many parts of the code we have to know whether the active partner is
+    the left partner, or the right partner. If the active partner is the
+    right partner we have to show the inverse name.
+
+    Because the active partner is crucial for the working of partner
+    relationships, we make sure on the res.partner model that the partner id
+    is set in the context where needed.
+    '''
+    _name = 'res.partner.relation'
+    _description = 'Partner relation'
+    _order = 'active desc, date_start desc, date_end desc'
+
+    def _get_computed_fields(
+            self, cr, uid, ids, field_names, arg, context=None):
+        '''Return a dictionary of dictionaries, with for every partner for
+        ids, the computed values.'''
+        def get_values(self, dummy_field_names, dummy_arg, context=None):
+            '''Get computed values for record'''
+            values = {}
+            on_right_partner = self._on_right_partner(self.right_partner_id.id)
+            # type_selection_id
+            values['type_selection_id'] = (
+                ((self.type_id.id) * 10) + (on_right_partner and 1 or 0))
+            # partner_id_display
+            values['partner_id_display'] = (
+                self.left_partner_id.id
+                if on_right_partner
+                else self.right_partner_id.id
+            )
+            return values
+
+        return dict([
+            (i.id, get_values(i, field_names, arg, context=context))
+            for i in self.browse(cr, uid, ids, context=context)
+        ])
+
+    _columns = {
+        'type_selection_id': osv.fields.function(
+            _get_computed_fields,
+            multi="computed_fields",
+            fnct_inv=lambda *args: None,
+            type='many2one', obj='res.partner.relation.type.selection',
+            string='Type',
+        ),
+        'partner_id_display': osv.fields.function(
+            _get_computed_fields,
+            multi="computed_fields",
+            fnct_inv=lambda *args: None,
+            type='many2one', obj='res.partner',
+            string='Partner'
+        ),
+    }
+
+    left_contact_type = fields.Selection(
+        lambda s: s.env['res.partner.relation.type']._get_partner_types(),
+        'Left Partner Type',
+        compute='_get_partner_type_any',
+        store=True,
+    )
+
+    right_contact_type = fields.Selection(
+        lambda s: s.env['res.partner.relation.type']._get_partner_types(),
+        'Right Partner Type',
+        compute='_get_partner_type_any',
+        store=True,
+    )
+
+    any_partner_id = fields.Many2many(
```

Comment:
From my understanding, this field is used to allow to search for a partner and see all of her relations. Does this actually work? It's not stored and I don't see a search function